UnicornEngine/video/include/unicorn/video/vulkan/Renderer.hpp

```diff
@@ -60,6 +64,13 @@ struct UniformCameraData
     glm::mat4 proj;
 };
 
+struct VkMaterial
+{
+    vk::DescriptorSet descriptorSet;
+    VkTexture* texture;
+    uint32_t nMeshes = 0; // Number of meshes, which point to this material
```

Comment:
`shared_ptr` instead of a counter?

Reply:
I thought about it, but where difference? Thread safety? But we haven't threads.

Reply:
Some one has to upgrade this counter and keep track of it. I'd rather have this logic encapsulated somewhere. Another reason would be that it looks like a classic example of shared ownership, which is a perfect case to use `shared_ptr`.